packages/bot-web-ui/src/pages/tutorials/guide-content/__tests__/guide-content.spec.tsx

```diff
@@ -47,21 +53,90 @@ describe('<FAQContent />', () => {
-            wrapper,
-        });
     });
 
-    it('should render the component', () => {
+    it('should render the GuideContent component', () => {
+        render(<GuideContent {...mocked_props} />, { wrapper });
+
         expect(screen.getByText('Step-by-step guides')).toBeInTheDocument();
         expect(screen.getByText('Videos on Deriv Bot')).toBeInTheDocument();
     });
+
+    it('should not define the active tour with function the setActiveTour() when it is the desktop version.', async () => {
+        mock_store.ui.is_desktop = true;
+        render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+
+        await waitFor(() => {
+            expect(mock_DBot_store.dashboard.active_tour).toBe('');
+        });
+    });
+
+    it('should open the dialog upon clicking the tutorial video button, should open the tab dashboard upon clicking tutorial card button.', async () => {
+        mock_store.ui.is_desktop = false;
+        const { container } = render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+
+        // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+        const tutorial_video_button = container.querySelector('.tutorials-wrap__placeholder__button-group--play');
+        userEvent.click(tutorial_video_button);
+
+        await waitFor(() => {
+            expect(mock_DBot_store?.dashboard.is_dialog_open).toBeTruthy();
+        });
+
+        const tutorial_card_button = screen.getAllByTestId('tutorials-wrap--tour')[0];
+        userEvent.click(tutorial_card_button);
+
+        await waitFor(() => {
+            expect(mock_DBot_store?.dashboard.setActiveTab(DBOT_TABS.DASHBOARD));
+        });
+    });
+
+    it('should trigger the keyDown event upon clicking "Enter"', async () => {
+        const mockEventListener = jest.fn();
+        render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+
+        document.addEventListener('keydown', mockEventListener);
+        const button_cancel = screen.getAllByTestId('tutorials-wrap--tour')[1];
+        // use fireEvent since userEvent doesn't handle the case: userEvent.type(button_cancel, '{Enter}');
+        fireEvent.keyDown(button_cancel, { key: 'Enter', code: 'Enter', keyCode: 13 });
+
+        await waitFor(() => {
+            expect(mockEventListener).toHaveBeenCalledWith(expect.objectContaining({ key: 'Enter', code: 'Enter' }));
+        });
+    });
+
+    it('should set active tab of bot builder when subtype is a "BotBuilder" and trigger tour button was clicked.', async () => {
+        render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+        const button_trigger_tour = screen.getAllByTestId('tutorials-wrap--tour')[1];
+        userEvent.click(button_trigger_tour);
+        await waitFor(() => {
+            expect(mock_DBot_store?.dashboard.setActiveTab(DBOT_TABS.BOT_BUILDER));
+        });
```

Comment:
@maryia-matskevich-deriv If you add another case as this one with `mock_store.ui.is_desktop = false;` and `expect(mock_DBot_store?.dashboard.setActiveTour('bot_builder'));` and
`expect(mock_DBot_store.dashboard.is_tour_dialog_visible).toBeTruthy();`
then it should cover the else cases in your test case

Reply:
@farabi-deriv sorry, I didn't get you, could you please describe it more clearly?

Reply:
All of the cases are covered. I'm confused🤨

Reply:
@maryia-matskevich-deriv your branches coverage is 85.71%. It is due to the else cases not being covered in this part. My suggestion was to help achieve fill 100& coverage.
<img width="530" alt="Screenshot 2024-08-27 at 11 10 05 AM" src="https://github.com/user-attachments/assets/a0544adc-340e-4715-b992-d03a23491010">

Reply:
@farabi-deriv ok, so what is the suggestion? where do you want to add those lines of code? it looks like all else conditions were passed

Reply:
The code above does not increase the coverage percentage, it works, but it is useless😒

Reply:
I have applied the suggested changes and it covers the else statements. The coverage also is now at 100%. Please try to avoid disregarding suggestions and calling it useless 😄 @maryia-matskevich-deriv 

<img width="639" alt="Screenshot 2024-08-27 at 4 28 58 PM" src="https://github.com/user-attachments/assets/bb35093e-eb04-4c7f-b03d-7a84a080e763">

Reply:
in private messages, we have found a way to improve it and refactor ✅ thank you @farabi-deriv for raising it😌